### PR TITLE
Render `<Button />` w/ `href` props as `<a />` to be `WAI-ARIA` compatible

### PR DESCRIPTION
**Pull request by shunkakinoki**
Thank you for the wonderful design system.

A quick PR to improve the accessibility of commonly used Button link patterns as well as address type configurability issues

### CODE REVIEW

**Comment by shunkakinoki** [comment]
Screenshot 2023-10-05 at 00 33 14
Wait oh.

**Comment by shunkakinoki** [comment]
Was it intentional

**Comment by fiveoutofnine** [comment]
no, ty for PR.

**Comment by fiveoutofnine** [comment]
WOW

**Comment by fiveoutofnine** [comment]
still requires a few things:
- `title: title || href || undefined` can be (and `title` would just be passed via `...rest`)
- `newTab` can be removed (and corresponding update throughout the project) + `onClick` can also just be passed via `...rest`
- same changes to `<IconButton />`
- update `/design/component/button` w/ new content/info

File: components/ui/button/index.tsx [hunk 1]
```diff
@@ -42,19 +43,23 @@ const Button = forwardRef(
       ...rest,
     };
 
+    // Destructure ref from props for Slot component to be type compatible
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { ref: _, ...otherProps } = props;
+
```
Comment (fiveoutofnine) [inline components/ui/button/index.tsx:49]:
```suggestion
    // Destructure `ref` from `props: JSX.IntrinsicElements['button']`, so the
    // remaining props are type-compatible with the `<Slot />` component
    // containing `<Link />` as its child.
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const { ref: _, ...rest } = props;

```

File: components/ui/button/index.tsx [hunk 2]
```diff
@@ -42,19 +43,23 @@ const Button = forwardRef(
...
     if (href && !newTab) {
       return (
-        <Link href={href} passHref legacyBehavior>
-          <button {...props}>
+        <Slot {...otherProps}>
```
Comment (fiveoutofnine) [inline components/ui/button/index.tsx:52]:
```suggestion
        <Slot {...rest}>
```

**Comment by fiveoutofnine** [comment]
to put it lightly, this is incredible. game changing. never seen anything like it before.

**Comment by fiveoutofnine** [comment]
https://github.com/waterfall-mkt/curta.wtf/pull/34

**Comment by fiveoutofnine** [comment]
implemented in #24 w/ fixes ~

**Pull request closed without merging**